Resources/MessageQueue/MQConnectionManager.py

```diff
@@ -0,0 +1,387 @@
+""" Class to manage connections for the Message Queue resources.
+    Also, set of 'private' helper functions to access and modify the message queue connection storage.
+    They are ment to be used only internally by the MQConnectionManager, which should
+    assure thread-safe access to it and standard S_OK/S_ERROR error handling.
+    MQConnection storage is a dict structure that contains the MQ connections used and reused for
+    producer/consumer communication. Example structure:
+    {
+      mardirac3.in2p3.fr: {'MQConnector':StompConnector, 'destinations':{'/queue/test1':['consumer1', 'producer1'],
+                                                                         '/queue/test2':['consumer1', 'producer1']}},
+      blabal.cern.ch:     {'MQConnector':None,           'destinations':{'/queue/test2':['consumer2', 'producer2',]}}
+    }
+"""
+
+from DIRAC import S_OK, S_ERROR, gLogger
+from DIRAC.Core.Utilities.LockRing import LockRing
+from DIRAC.Resources.MessageQueue.Utilities import getMQService
+from DIRAC.Resources.MessageQueue.Utilities import getDestinationAddress
+from DIRAC.Resources.MessageQueue.Utilities import createMQConnector
+import re
+
+
+class MQConnectionManager(object):
+  """Manages connections for the Message Queue resources in form of the interal connection storage."""
+  def __init__(self, connectionStorage = None):
+    self.log = gLogger.getSubLogger( self.__class__.__name__ )
+    self._lock = None
+    if connectionStorage:
+      self._connectionStorage = connectionStorage
```

Comment:
According to this https://docs.python.org/3.4/tutorial/classes.html#tut-private and  https://www.python.org/dev/peps/pep-0008/
Python convention says that everything starting with single underscore is non-public (or non-API), 
and for double underscore: "If your class is intended to be subclassed, and you have attributes that you do not want subclasses to use, consider naming them with double leading underscores and no trailing underscores." ()
so _ would be like protected and __ like private indeed
But in this case neither I want to subclass this class nor I want forbid the access to e.g connectionStorage
It's a convention anyway, so If we use in DIRAC  double underscore then I can change it.